internal/http/routes/api/auth.go

```diff
@@ -0,0 +1,85 @@
+package api
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-shiori/shiori/internal/config"
+	"github.com/go-shiori/shiori/internal/http/response"
+	"github.com/gofiber/fiber/v2"
+	"go.uber.org/zap"
+)
+
+type AuthAPIRoutes struct {
+	logger *zap.Logger
+	router *fiber.App
+	deps   *config.Dependencies
+}
+
+func (r *AuthAPIRoutes) Setup() *AuthAPIRoutes {
+	r.router.Post("/login", r.loginHandler)
```

Comment:
I think it's a little veiled to declare route here. I prefer the original way, put routes together and use difference handlers

Reply:
The problem with using a single package for everything is that after a while you either have huge files containing all kinds of related logic for the API (handlers, payloads, responses, middlewares, codecs, ...). I followed a similar approach here as with the "domains", each endpoint is contained in a "domain", in this case this is the "auth" domain, so all routes for the auth domain are defined here, and all domains are defined in _api.go_. (auth, bookmarks, tags, ...). As with everything, suggestions welcome!

Reply:
I get you.I think it's just a matter of taste. I don't think I'm much experienced in dev in go. I can learn from this :)